docs/models/validations-and-constraints.md

```diff
@@ -0,0 +1,243 @@
+import { Attribute, NotNull, ValidateAttribute } from '@sequelize/core/decorators-legacy';
+
+class User extends Model {
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  // highlight-start
+  @ValidateAttribute((value: unknown, user: User, attributeName: string) => {
+    // this function will run when this attribute is validated.
+    if (name.length === 0) {
+      throw new Error('Name cannot be empty');
+    }
+  })
+  // highlight-end
+  declare name: string;
+}
+```
+
+### `@sequelize/validator.js`
+
+The [`@sequelize/validator.js`](https://www.npmjs.com/package/@sequelize/validator.js) package provides a number validators 
+based on the [`validator.js`](https://www.npmjs.com/package/validator) package, such as email validation and regex matching.
+
+__⚠️ As indicated in the validator.js documentation, the library validates and sanitizes strings only.__
+
+```ts
+import { Model, DataTypes } from '@sequelize/core';
+import { Attribute, NotNull } from '@sequelize/core/decorators-legacy';
+import { IsEmail } from '@sequelize/validator.js';
+
+class User extends Model {
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  // highlight-next-line
+  @IsEmail
+  declare email: string;
+}
+```
+
+Head to [its TypeDoc page](pathname:///api/v7/modules/_sequelize_validator_js.html) for the list of available validators.
+
+:::note Future development
+
+We're working on adding support for more validation libraries, See [issue #15497](https://github.com/sequelize/sequelize/issues/15497).
+
+:::
+
+### Model validators
+
+Validations can also be defined to check the model after the attribute-specific validators.
+
+Using this you could, for example, ensure either neither of `latitude` and `longitude` are set or both, and fail if one but not the other is set.
+
+Model validator methods are called with the model object's context and are deemed to fail if they throw an error, otherwise pass. This is just the same as with custom attribute-specific validators.
+
+Any error messages collected are put in the validation result object alongside the attribute validation errors, with keys named after the failed validation method's key in the `validate` option object. Even though there can only be one error message for each model validation method at any one time, it is presented as a single string error in an array, to maximize consistency with the field errors.
+
+An example:
+
+```ts
+class Place extends Model {
+  @Attribute(DataTypes.INTEGER)
+  declare latitude: number | null;
+
+  @Attribute(DataTypes.INTEGER)
+  declare longitude: number | null;
+  
+  // highlight-start
+  @ModelValidator
+  static validateCoords() {
+    if ((this.latitude === null) !== (this.longitude === null)) {
+      throw new Error('Either both latitude and longitude, or neither!');
+    }
+  }
+  // highlight-end
+}
+```
+
+This will ensure that if either `latitude` or `longitude` is set, then both are set, and if neither is set, then neither is set.
```

Comment:
```suggestion
This will ensure that if either `latitude` or `longitude` is set, then both are set. If neither is set, then neither is set.
```

Unclear. And, is this explanation required?